codecarbon/external/hardware.py

```diff
@@ -90,6 +102,43 @@ def _get_power_from_cpus(self) -> Power:
                 power += value
         return Power.from_watts(power)
 
+    def _parse_cpu_model(self, raw_name) -> str:
+        """
+        Parse the model name from the raw name extracted from cpuinfo library
+        :return: parsed CPU name
+        """
+        return (
+            raw_name.split(" @")[0]
+            .replace("(R)", "")
+            .replace("(TM)", "")
+            .replace(" CPU", "")
+        )
+
+    def _get_power_from_constant(self) -> Power:
+        """
+        Get CPU power from constant mode
+        :return: power in KW
+        """
+        cpu_info = cpuinfo.get_cpu_info()
+        if cpu_info:
+            model_raw = cpu_info["brand_raw"]
+            model = self._parse_cpu_model(model_raw)
+            cpu_power_df = DataSource().get_cpu_power_data()
```

Comment:
Does this mean that the 2000+line CSV will be read and parsed _every time_ the cpu power is required? Like every 15seconds? We should have that cached no?

Reply:
I think it's now only happening when creating the EmissionTracker object, could you confirm @QinganZhao?

Reply:
Yes - I made the changes hours ago